v2/tools/generator/internal/codegen/pipeline/add_secrets.go

```diff
@@ -54,7 +53,7 @@ func applyConfigSecretOverrides(config *config.Configuration, definitions astmod
 
 	applyConfigSecrets := func(_ *astmodel.TypeVisitor, it *astmodel.ObjectType, ctx interface{}) (astmodel.Type, error) {
 		typeName := ctx.(astmodel.TypeName)
-		for _, prop := range it.Properties() {
+		for _, prop := range it.Properties().Copy() {
```

Comment:
why do we need `.Copy()` here?

Reply:
`Properties` is a `ReadOnlyPropertySet` which you cannot iterate over. `Copy`ing it is pessimistic but also what the existing behaviour was, so it doesn’t make performance _worse_.

Reply:
Ahh gotcha, I missed that. That's fine then carry on.